images/scripts/virt-install-rhel4edge

```diff
@@ -0,0 +1,132 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+# Adapted from create-guest-qcow2.  We don't use that script as is
+# since it is, to quote Mr Parker, "2% code and 98% config stuff".
+
+set -ex
+
+if [ "$#" -lt 3 ]; then
+    echo >&2 "Usage: virt-install-rhel4edge IMAGE ARCH URL"
+    exit 1
+fi
+
+out=$1
+arch=$2
+url=$3
+
+base=$(dirname $0)
+ks=fedora.ks
+
+if [ -e "$NODE_EXTRA_CA_CERTS" ]; then
+    crt="$(cat "$NODE_EXTRA_CA_CERTS")"
+    crt_command="echo '$crt' > /root/node_extra_ca_certs"
+fi
+
+cat <<EOF >$ks
+text
+lang en_US.UTF-8
+keyboard us
+timezone Etc/UTC --isUtc
+zerombr
+clearpart --all --initlabel
+network --bootproto dhcp
+rootpw foobar
+firewall --enabled --ssh
+autopart
+shutdown
+user --name=admin --groups=wheel --password=foobar
+ostreesetup --osname=rhel --url=file:///run/install/repo/ostree/repo --ref=rhel/9/x86_64/edge --nogpg
+
+%post
+mkdir /root/.ssh
+chmod 700 /root/.ssh
+echo "$(cat $base/../../machine/identity.pub)" > /root/.ssh/authorized_keys
+
+systemctl is-enabled sshd.socket || systemctl is-enabled sshd.service || systemctl enable sshd.socket
+%end
+
+EOF
+
+qemu-img create -f qcow2 "$out" 12G
+
+connect="--connect=qemu:///session"
+name=$(basename $1)-builder
+
+cleanup_builder_vm () {
+  if virsh $connect list --state-running --name | grep -q $name; then
+      virsh $connect destroy $name
+  fi
+  if virsh $connect list --all --name | grep -q $name; then
+      virsh $connect undefine $name
+  fi
+}
+
+# Using virt-install is fundamentally an interactive process.  We
+# expect the installation to finish unattended, but it might drop into
+# an emergency shell upon unexpected errors, for example.  Thus, we
+# run it under 'expect' and catch a few known scenarios that require
+# special handling.
+#
+# (Also, virt-install works best when it has a tty.)
+
+supervised_virt_install () {
+expect -- - virt-install "$@" <<'EOF'
+eval spawn $argv
+set timeout -1
+expect {
+  "emergency mode" { send_user "\n\n\n\nABORT - emergency shell\n"
+                     exit 1
+                   }
+  "Please make your choice" { send_user "\n\n\n\nABORT - Anaconda stopped\n"
+                              exit 1
+                            }
+  "Please make a selection" { send_user "\n\n\n\nABORT - Anaconda stopped\n"
+                              exit 1
+                            }
+}
+EOF
+}
+
+cleanup_builder_vm
+
+
+supervised_virt_install $connect \
+                        --name=$name \
+                        --initrd-inject=$ks \
+                        --extra-args="inst.ks=file:/$ks console=ttyS0,115200" \
+                        --disk "path=$out,format=qcow2" \
+                        --ram 4096 \
+                        --vcpus=1 \
+                        --os-variant fedora21 \
```

Comment:
we should *really* bump that :grin:

Reply:
actually default is `--os-variant detect=on` which works just fine, dropped it